Site/dataobjects/SiteApiCredential.php

```diff
@@ -79,6 +79,35 @@ public function loadByApiKey($key, SiteInstance $instance =  null)
 		return true;
 	}
 
+	// }}}
+	// {{{ public function getPromotionByType()
+
+	public function getPromotionByType($type)
+	{
+		$this->checkDB();
+
+		$sql = sprintf(
+			'SELECT *
+			FROM %s
+			INNER JOIN Promotion ON
+				Promotion.api_credential = %s.id
+			WHERE %s.id = %s
+				AND Promotion.api_sign_on_type = %s',
+			$this->table,
+			$this->table,
+			$this->table,
+			$this->db->quote($this->id, 'integer'),
+			$this->db->quote($type, 'text')
+		);
+
+		// There should only be one type in the database
+		return SwatDB::query(
+			$this->db,
+			$sql,
+			SwatDBClassMap::get('RapPromotionWrapper')
```

Comment:
The Site package should not reference Rap code. Site does not know about promotion objects. This method should move to the `silverorange/kos` package.